metricbeat/module/coredns/stats/_meta/data_request_duration_sec_event.json

```diff
@@ -0,0 +1,54 @@
+{
+    "@timestamp": "2017-10-12T08:05:34.853Z",
+    "agent": {
+        "hostname": "host.example.com",
+        "name": "host.example.com"
+    },
+    "coredns": {
+        "stats": {
+            "dns": {
+                "request": {
+                    "duration": {
+                        "sec": {
+                            "bucket": {
+                                "+Inf": 2210,
+                                "0.00025": 0,
+                                "0.0005": 0,
+                                "0.001": 0,
+                                "0.002": 0,
+                                "0.004": 0,
+                                "0.008": 0,
+                                "0.016": 0,
+                                "0.032": 0,
+                                "0.064": 0,
+                                "0.128": 2200,
+                                "0.256": 2210,
+                                "0.512": 2210,
+                                "1.024": 2210,
+                                "2.048": 2210,
+                                "4.096": 2210,
+                                "8.192": 2210
```

Comment:
Are these fields names (`0.00025`, `0.0005`...) fixed quantities? If not we'd have to think in another way of storing them to avoid fields explossion.

In any case you will need to remove the dots.

Reply:
Yes, these are fixed buckets coming from Prometheus plugin of Coredns:
- [hist1](https://github.com/coredns/coredns/blob/b7480d5d1216aa87d80d240a31de750079eba904/plugin/metrics/vars/vars.go#L31)
- [hist2](https://github.com/coredns/coredns/blob/b7480d5d1216aa87d80d240a31de750079eba904/plugin/metrics/vars/vars.go#L53)
- [hist3.1](https://github.com/coredns/coredns/blob/b7480d5d1216aa87d80d240a31de750079eba904/plugin/metrics/vars/vars.go#L22) [hist3.2](https://github.com/coredns/coredns/blob/b7bbfef6ef3d057436df0dde25ceb3f31ee9309d/plugin/plugin.go#L105)

Regarding the dots, do you think we should change the prometheus wrapper to handle this? :thinking:

Reply:
> Yes, these are fixed buckets coming from Prometheus plugin of Coredns:

:+1: 

> Regarding the dots, do you think we should change the prometheus wrapper to handle this? thinking

Maybe yes. I am not sure how we could do it for this case, but we shouldn't generate field names with dots as they are stored as objects in elasticsearch.

I mean this:
```
{
  "bucket": {
    "0.001": 0,
    "0.002": 0,
  }
}
```
Is stored as:
```
{
  "bucket": {
    "0": {
      "001": 0,
      "002": 0,
    }
  }
}
```

What can be problematic, for example we couldn't store:
```
{
  "bucket": {
    "0": 1.0,
    "0.002": 0,
  }
}
```
Because `bucket.0` would need to be a number (`1.0`) and an object (`{ "002": 0 }`) at the same time.

Reply:
Maybe an option here could be to convert these seconds to nanosecods so they can be represented as integers. We'd have to see how to add an option to the prometheus helpers for this.

Reply:
:+1: